flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/FlinkRuleSets.scala

```diff
@@ -49,6 +51,15 @@ object FlinkRuleSets {
   val POST_EXPAND_CLEAN_UP_RULES: RuleSet = RuleSets.ofList(
     EnumerableToLogicalTableScan.INSTANCE)
 
+  /**
+    * Pre-processing for logical optimization.
+    */
+  val PRE_LOGICAL_OPT_RULES: RuleSet = RuleSets.ofList(
+    new FlinkAggregateExtractProjectRule(
+      operand(classOf[Aggregate],
+        operand(classOf[RelNode], none)), RelFactories.LOGICAL_BUILDER),
+    ReduceExpressionsRule.PROJECT_INSTANCE)
```

Comment:
only project or also filters? do we still need the same rule in other phase? currently it is also used in `*_NORM_RULES`.

Reply:
I will put the extract rule into the normalization phase, so we don't have the problem here.